app/features/sidebar/pools-section/pool-item.tsx

```diff
@@ -45,38 +47,40 @@ const StyledPoolItem = styled(StyledItem)<{isSelected: boolean}>`
   }
 `
 
-const PoolItem = ({innerRef, styles, data: pool, state, handlers}) => {
+// Is this duplicated somewhere?
+const PoolItem = ({innerRef, styles, data, state, handlers}) => {
+  const pool = data as Pool
   const dispatch = useDispatch<AppDispatch>()
   const workspaceId = useSelector(Current.getWorkspaceId)
   const currentPoolId = useSelector(Current.getPoolId)
+  const ingest = useSelector(Imports.get(currentPoolId))
   const history = useHistory()
   const {isEditing} = state
-  const p = brim.pool(pool)
   const ctxMenu: MenuItemConstructorOptions[] = [
     {
       label: "Rename",
       click: () => {
         handlers.edit()
       }
     },
-    ...dispatch(getPoolContextMenu(p))
+    ...dispatch(getPoolContextMenu(pool))
   ]
 
   const onClick = (e) => {
     e.preventDefault()
     history.push(
-      lakeSearchPath(p.id, workspaceId, {
-        spanArgs: p.empty() ? undefined : p.defaultSpanArgs()
+      poolSearchPath(pool.id, workspaceId, {
+        spanArgs: pool.empty() ? undefined : pool.defaultSpanArgs()
       })
     )
   }
 
-  const progress = p.ingesting() && (
+  const progress = ingest && isNumber(ingest.progress) && (
     <div className="small-progress-bar">
-      <ProgressIndicator percent={p.ingestProgress()} />
+      <ProgressIndicator percent={ingest.progress} />
```

Comment:
Got it, so ingest is no longer stored in the pool. That certainly makes updating that particular slice in redux simpler 👌

Reply:
Exactly! Although I need to rename the state slice to "Ingests". The reason is because you can name a variable "import". It's a keyword!

```
const import = Imports.get(id) // wont work!
```